gwctl/pkg/resourcediscovery/nodes.go

```diff
@@ -246,11 +247,15 @@ func (b *BackendNode) ID() backendID {
 	)
 }
 
-// HTTPRouteNode models the relationships and dependencies of a Namespace.
+// NamespaceNode models the relationships and dependencies of a Namespace.
 type NamespaceNode struct {
 	// NamespaceName identifies the Namespace.
 	NamespaceName string
```

Comment:
Lets replace the NamespaceName with the actual `corev1.Namespace` resource. This would make this node consistent with how the other nodes are defined -- where nodes are simply wrappers over the underlying base resource. This would also mean that we don't need to define each field of the Namepace individually in the node here.

This means that now:
1. We can modify the `ResourceModel.addNamespace()` to accept `namespaces ...corev1.Namespace` and not need a separate `addNamespaceWithLabelsAnnotationsStatus`
2. Within `Discoverer.discoverNamespaces()`, we would fetch all `corev1.Namespace` resources. Then, while iterating over the existing resources, once we figure out the namespace of the resource, we call `addNamespace()` and pass in the `corev1.Namespace` that we would have already fetched.

Reply:
The reason I created a seperate `addNamespaceWithLabelsAnnotationsStatus` function was because in `Discoverer.discoverNamespaces()` we have `resourceModel.addNamespace()` for gateway, httproute and backends which already return string to the function, so I were to modify `ResourceModel.addNamespace()` to accept `namespaces ...corev1.Namespace` then those function calls would break.
How can I handle those functions?